Library/Homebrew/livecheck/strategy.rb

```diff
@@ -52,19 +52,28 @@ def from_symbol(symbol)
       # Returns an array of strategies that apply to the provided URL.
       #
       # @param url [String] the URL to check for matching strategies
-      # @param regex_provided [Boolean] whether a regex is provided in a
+      # @param livecheck_strategy [Symbol] a {Strategy} symbol from the
+      #   `livecheck` block
+      # @param regex_provided [Boolean] whether a regex is provided in the
       #   `livecheck` block
       # @return [Array]
-      def from_url(url, regex_provided = nil)
+      def from_url(url, livecheck_strategy: nil, regex_provided: nil)
         usable_strategies = strategies.values.select do |strategy|
-          # Ignore strategies with a priority of 0 or lower
-          next if strategy.const_defined?(:PRIORITY) && !strategy::PRIORITY.positive?
+          if strategy == PageMatch
+            # Only treat the `PageMatch` strategy as usable if a regex is
+            # present in the `livecheck` block
+            next unless regex_provided
+          elsif strategy.const_defined?(:PRIORITY) &&
+                !strategy::PRIORITY.positive? &&
+                from_symbol(livecheck_strategy) != strategy
+            # Ignore strategies with a priority of 0 or lower, unless the
+            # strategy is specified in the `livecheck` block
```

Comment:
We explicitly state on line 17 of this file that strategies with priorities of 0 or lower are ignored.
Shall we expand and say that "negative priorities raise an error (they don't at the moment, but they probably should), 0 priority is reserved for..., all normal strategies should use positive values below 10" ?

Why do we allow such strategies when they're specified in the `livecheck` block? I guess this comment should explain why we're doing this instead of or in addition to what it currently says.

Reply:
As this comment mentions, the 0-10 range is informal. It's technically possible for us to go above or below this range if it becomes necessary in the future and that's why we don't strictly enforce the range in the code.

The only ideas that we strictly enforce in the code are:

* The default priority is 5
* Non-positive priorities (0 or below) are omitted unless specified in the `livecheck` block
* Strategies that apply to a URL are arranged from highest priority to lowest

I do think this explanation can be improved a little (e.g., I didn't mention here that the way to apply a strategy with a non-positive priority is using `strategy` in a `livecheck` block), so I'll take care of that in a moment.